src/per_project_setup.rs

```diff
@@ -0,0 +1,117 @@
+//! use the LLVM linker and only GNU for `objdump`, but the GNU `objdump` won't
+//! be able to show all info if you mix utilities like this, because (i guess?)
+//! the GNU `objdump` doesn't understand LLVM linker info format. Using the GNU
+//!  linker will make sure you can see all the debug info. I've never noticed a
+//! linking speed difference between LLVM and GNU on the GBA, programs just
+//! don't get that big. Put `-Clinker=arm-none-eabi-ld` into the `rustflags` for
+//! our target and we'll get the GNU linker.
+//!
+//! Regardless of if you're using the LLVM or GNU linker, you'll need a linker
+//! script. The [Github Repo][github_script] for this project has a usable
+//! linker script. You should save this file into your own project, such as in a
+//! folder called `linker_scripts/` under the name `mono_boot.ld`. Then we add a
+//! `rustflags` argument with `-Clink-arg=-T<path>`, using the script's path
+//! relative to our project root.
+//!
+//! You can also set `-Ctarget-cpu=arm7tdmi`, because that is the specific CPU
+//! of the GBA, though I'm not sure that LLVM really does much with the
+//! information. In my very limited testing, I didn't see any different code
+//! generated when setting the CPU compared to not doing it, but I do it just in
+//! case it helps somehow, because it doesn't hurt.
+//!
+//! [github_script]:
+//!     https://github.com/rust-console/gba/blob/main/linker_scripts/mono_boot.ld
+//!
+//! Your `.cargo/config.toml` should probably look like the following:
+//! ```toml
+//! [build]
+//! target = "thumbv4t-none-eabi"
+//!
+//! [unstable]
+//! build-std = ["core"]
+//!
+//! [target.thumbv4t-none-eabi]
+//! runner = "mgba-qt"
+//! rustflags = [
+//!   "-Clinker=arm-none-eabi-ld",
+//!   "-Clink-arg=-Tlinker_scripts/mono_boot.ld",
+//!   "-Ctarget-cpu=arm7tdmi",
+//! ]
+//! ```
+//!
+//! * If you want the `compiler_builtins` crate to provide weak intrinsics so
+//!   that you can override them yourself, you can set `build-std-features =
+//!   ["compiler-builtins-weak-intrinsics"]` in the `[unstable]` section. This
+//!   is not normally needed, but if you some day try to write your own
+//!   intrinsics impls you'll want to know it's available.
+//!
+//! ## Rust Analyzer
+//!
+//! The `test` crate won't be available when using `build-std` like this. To
+//! make Rust Analyzer work without constantly complaining about a missing
+//! `test` crate, add this to your `.vscode/settings.json` (or whatever similar
+//! file in your editor of choice)
+//!
+//! ```json
+//! {
+//!   "rust-analyzer.cargo.allTargets": false,
+//!   "rust-analyzer.check.command": "build",
+//!   "rust-analyzer.check.extraArgs": [
+//!     "--lib",
+//!     "--bins",
+//!     "--examples"
+//!   ]
+//! }
+//! ```
+//!
+//! ## `no_std` and `no_main`
+//!
+//! The full standard library isn't available on the GBA, it's a [bare metal]
+//! environment, so we'll only have access to the [`core`] crate.
```

Comment:
Maybe mention `alloc` + give hints about where to go when looking to set it up? I don't use it myself, but I gather that it's common to place the heap in ewram. Something like [`linked_list_allocator`](https://crates.io/crates/linked_list_allocator) seems like a good choice.